_source/_posts/2022-05-13-express-prisma-and-tailwindcss.md

```diff
@@ -0,0 +1,665 @@
+
+[Prisma](https://www.prisma.io/) is an ORM for Node.js using TypeScript.  It integrates with many of the most popular databases today including MySQL, SQL Server, SQLite, and MongoDB.  CockroachDB is available in preview mode.  Prisma emphasizes a human readable schema with a type-safe database client.  Prisma also includes other features such as migrations, seed data, and a virtual database browser.  
+
+In this project, you will use Prisma to connect to a database server.  You will build a schema to model a workout tracker application.  Then you will create some seed data to begin working with and use Prisma to migrate and seed your database.  You will then create the workout web application using [Pug](https://pugjs.org/api/getting-started.html) and [tailwindcss](https://tailwindcss.com/) on your front end to compliment your express backend.  
+
+### Prequresites
+
+To continue you will need:
+
+- An IDE for javascript.  I will use [Visual Studio Code](https://code.visualstudio.com/) but you can use Atom, Webstorm, or any other IDE you wish
+- [Node.js](https://nodejs.org/en/)
+- A database, one of PostgreSQL, MySQL, SQLite, SQL Server, or MongoDB.  I will be using SQL Server
+- An [Okta Developer Account](https://developer.okta.com/) (free forever, to handle your OAuth needs)
+- [Okta CLI](https://cli.okta.com)
+
+If you wish to look at the app you can [view it on github](https://github.com/nickolasfisher/Okta_Primsa).
+
+## Create your Okta application
+
+Create a new directory for your application.  
+
+{% include setup/cli.md type="web" loginRedirectUri="http://localhost:3000/authorization-code/callback" %}
+
+## Create your Express Application
+
+Next, you will use the `express-generator` application generator tool to quickly scaffold your application.  Run the following command.
+
+```console
+npx express-generator --view=pug
+```
+
+Next, you will need to install your packages.  
+
+```console
+npm i @prisma/client@3.13.0
+npm i dotenv@16.0.0
+npm i passport@0.5.2
+npm i passport-openidconnect@0.1.1
+npm i express-session@1.17
+npm i -D tailwindcss@3.0.24
+npm i -D prisma@3.13.0
+```
+
+> `@prisma/client` is used to access the database from your server code
+> `dotenv` reads configuration settings from `.env` files like the one produced by the Okta CLI
+> `passport` is a middleware for Node.js that is flexible enough to handle most authentication scenarios, including Okta.  `passport-openidconnect` is a module for passport that lets you authenticate with OpenID Connect
+> `express-session` is required for passport.
+> `tailwindcss` is the CSS framework you will use.  If you've never used tailwindcss before you may wonder why it's a development dependency.  This is because tailwindcss will dynamically build your CSS files from your views and configuration.  More on that later
```

Comment:
```suggestion
> `Tailwind CSS` is the CSS framework you will use.  If you've never used Tailwind before, you may wonder why it's a development dependency.  This is because Tailwind will dynamically build your CSS files from your views and configuration.  More on that later
```